### PR TITLE
adds protoc-gen-[prost, prost-crate, prost-serde, tonic]

### PR DESCRIPTION
**Pull request by SitAAAx**
###### Description of changes

protoc-gen-prost: init at 0.2.3
protoc-gen-prost-crate: init at 0.3.1
protoc-gen-prost-serde: init at 0.2.3
protoc-gen-tonic: init at 0.3.0

Protobuf plugins for generating Rust (inclusive gRPC) code.

The code of all mentioned packages lives at: https://github.com/neoeinstein/protoc-gen-prost
Their crates - which are used for packaging - come from:
https://crates.io/crates/protoc-gen-prost
https://crates.io/crates/protoc-gen-prost-crate
https://crates.io/crates/protoc-gen-prost-serde
https://crates.io/crates/protoc-gen-tonic

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/protoc-gen-prost-crate/default.nix [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+  cargoSha256 = "sha256-dcKJRX/iHIWEmBD2nTMyQozxld8b7dhxxB85quPUysg=";
+
+  cargoBuildFlags = ["--bin" pname];
+
+  nativeBuildInputs = [ protobuf ];
+
+  meta = with lib; {
+    description = "Protocol Buffers compiler plugin powered by Prost!";
+    homepage = "https://github.com/neoeinstein/protoc-gen-prost";
```
Comment (felschr) [inline pkgs/development/tools/protoc-gen-prost-crate/default.nix:24]:
```suggestion
    homepage = "https://github.com/neoeinstein/protoc-gen-prost";
    changelog = "https://github.com/neoeinstein/protoc-gen-prost/blob/main/CHANGELOG.md";
```

and for the other packages as well.

File: pkgs/development/tools/protoc-gen-prost-crate/default.nix [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  version = "0.3.1";
+
+  src = fetchCrate {
+    inherit pname version;
+    sha256 = "sha256-MtGeU2PnVYPXb3nly2UaryjmjMz1lxcvYDjFiwf58FA=";
+  };
+
+  cargoSha256 = "sha256-dcKJRX/iHIWEmBD2nTMyQozxld8b7dhxxB85quPUysg=";
+
+  cargoBuildFlags = ["--bin" pname];
```
Comment (felschr) [inline pkgs/development/tools/protoc-gen-prost-crate/default.nix:18]:
```suggestion
  cargoBuildFlags = [ "--bin" pname ];
```

Just to keep the style consistent within the file.

**Review by felschr — CHANGES_REQUESTED** [review]
I'd like to see these plugins in nixpkgs, are you still interested and willing to update the PR to the latest package versions?

Please split up the commits into one for changing the maintainer list & one for each package following the format described in [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md):
`maintainers: add sitaaax`
`protoc-gen-prost: init at 0.2.3`, etc.

You can also add me as maintainer as I'm interested in keeping these up-to-date & working.

**Comment by SitAAAx** [comment]
> Please split up the commits into one for changing the maintainer list & one for each package following the format described in [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md): `maintainers: add sitaaax` `protoc-gen-prost: init at 0.2.3`, etc.

Sorry, saw this comment after pushing the updates. Will change it after finishing my day job.

> You can also add me as maintainer as I'm interested in keeping these up-to-date & working.

Sure, will add you.

**Comment by SitAAAx** [comment]
Restructured commits hopefully as requested.
Removed `cargoBuildFlags` and `nativeBuildInputs` because they do not seem to have an effect on the output.
Used back then https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/tools/protoc-gen-rust/default.nix as template.
But I am still new to nix packaging and can not say if I am doing something wrong.
Please bear with me.

**Comment by felschr** [comment]
Result of `nixpkgs-review pr 203836` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>protoc-gen-prost</li>
    <li>protoc-gen-prost-crate</li>
    <li>protoc-gen-prost-serde</li>
    <li>protoc-gen-tonic</li>
  </ul>
</details>

**Review by felschr — APPROVED** [review]
Looks good to me.
I've tested the `protoc-gen-prost-crate` plugin with buf code generation and it works as expected.

**Pull request merged**